### PR TITLE
Starting work on documentation

### PR DESCRIPTION
**Pull request by TheMcMurder**
Initial documentation for Constructing routes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,91 @@
 [![Build Status](https://travis-ci.com/single-spa/single-spa-layout.svg?branch=master)](https://travis-ci.com/single-spa/single-spa-layout)
 
 Layout engine for single-spa applications
+
+## Project Goal
+Single-spa layout is trying to solve the following problems:
```
Comment (frehner) [inline README.md:8]: would just say `solves` or `attempts to solve` so we don't have to make a new PR to update this in the future :D

File: README.md [hunk 2]
```diff
@@ -3,3 +3,91 @@
...
+applications.forEach(registerApplication);
+const layoutEngine = constructLayoutEngine({routes, applications})
+start()
+```
+
+
+### Constructing Routes
+single-spa-layout supports two different sytanxes defining applications and routes at this time:
+- JSX (compiles to JSON syntax)
+- JSON
```
Comment (frehner) [inline README.md:40]: can we link to the sections below here?

File: README.md [hunk 3]
```diff
@@ -3,3 +3,91 @@
...
+    },
+    {
+      type: "route",
+      path: "settings",
+      routes: [{ type: "application", name: "@org/settings" }],
+    },
+    { type: "application", name: "@org/footer" },
+  ],
+}
+```
```
Comment (TheMcMurder) [inline README.md:93]: I'm pretty sure the examples I've put here are equivalent. 

Next steps:
- Document the options that each object can have during `constructRoute` phase
- Document `constructApplications`

**Comment by joeldenning** [comment]
The documentation will need to move over to single-spa.js.org once we release, but this is good to have for now.

**Pull request merged**